### PR TITLE
fix MacOS host mode docker host IP detection

### PR DESCRIPTION
**Pull request by alexrashed**
This PR extends the endpoint detection (i.e. detecting the IP address of LocalStack) in host mode (outside of docker) for other systems than Linux.
This is a draft PR, and I would love to get further input and feedback!
/cc @dfangl @dominikschubert @joe4dev @simonrw

### CODE REVIEW

File: localstack/utils/container_networking.py [hunk 1]
```diff
@@ -69,15 +70,30 @@ def get_endpoint_for_network(network: Optional[str] = None) -> str:
             )
         else:
             # default gateway for the network should be the host
-            # (only under Linux - otherwise fall back to DOCKER_HOST_FROM_CONTAINER below)
+            # (only under Linux - otherwise we need to determine the IP by starting a container below)
```
Comment (joe4dev) [inline localstack/utils/container_networking.py:73]: suggestion for clearer comment:

```
# In a Linux host-mode environment, the default gateway for the network should be the IP of the host
```

File: localstack/utils/container_networking.py [hunk 2]
```diff
@@ -69,15 +70,30 @@ def get_endpoint_for_network(network: Optional[str] = None) -> str:
...
             if config.is_in_linux:
                 main_container_ip = DOCKER_CLIENT.inspect_network(network)["IPAM"]["Config"][0][
                     "Gateway"
                 ]
+            else:
+                # we are not in docker and not on linux, we need to determine the IP of the host by running a container
```
Comment (joe4dev) [inline localstack/utils/container_networking.py:79]:
```
# In a non-Linux host-mode environment, we need to determine the IP of the host by running a container
# (basically MacOS host mode, i.e. this is a feature to improve the developer experience)
```

**Comment by joe4dev** [comment]
`get_endpoint_for_network` is used in:
* populate the `LOCALSTACK_HOSTNAME` environment variable for the Lambda container (`localstack.services.awslambda.invocation.runtime_environment.RuntimeEnvironment.get_environment_variables`)
* populate `LOCALSTACK_HOSTNAME` in ECS (`localstack_ext.services.ecs.provider.ECSProvider._do_run_task`)
* populate `LOCALSTACK_HOSTNAME` in bigdata (`localstack_ext.utils.bigdata.bigdata_utils.BigdataServersSeparateContainer.start_spark_container`)
* populate `$LOCALSTACK_HOSTNAME` in Batch (`localstack_ext.services.batch.provider.BatchProvider.prepare_submit_job_request`)

I am a little surprised that not more usages show up (e.g., EC2, StepFunctions) 🤔

Doing this lookup dynamically using a cached method speeds up container startup (e.g., in the CLI scenario).

**Review by joe4dev — APPROVED** [review]
Tested on macOS ✅ 

Test case: `tests.integration.awslambda.test_lambda.TestLambdaFeatures.test_invocation_type_request_response`

Tested against community and pro.

**Review by dfangl — APPROVED** [review]
I think this is fine, especially since it only affects host mode anyway. Just have a minor suggestion to avoid spawning a bash shell (and potentially be a couple of cycles faster).

**Pull request merged**